mmcv/image/io.py

```diff
@@ -239,20 +260,57 @@ def imfrombytes(content, flag='color', channel_order='bgr', backend=None):
         return img
 
 
-def imwrite(img, file_path, params=None, auto_mkdir=True):
+def imwrite(img,
+            file_path,
+            params=None,
+            auto_mkdir=None,
+            file_client_args=None):
     """Write image to file.
 
+    Note:
+        In v1.4.1 and later, add `file_client_args` parameters.
+
+    Warning:
+        The parameter `auto_mkdir` will be deprecated in the future and every
+        file clients will make directory automatically.
+
     Args:
         img (ndarray): Image array to be written.
         file_path (str): Image file path.
         params (None or list): Same as opencv :func:`imwrite` interface.
         auto_mkdir (bool): If the parent folder of `file_path` does not exist,
-            whether to create it automatically.
+            whether to create it automatically. It will be deprecated.
+        file_client_args (dict | None): Arguments to instantiate a
+            FileClient. See :class:`mmcv.fileio.FileClient` for details.
+            Default: None.
 
     Returns:
         bool: Successful or not.
+
+    Examples:
+        >>> # write to hard disk client
+        >>> ret = mmcv.imwrite(img, '/path/to/img.jpg')
+        >>> # infer the file backend by the prefix s3
+        >>> ret = mmcv.imwrite(img, 's3://bucket/img.jpg')
+        >>> # manually set the file backend petrel
+        >>> ret = mmcv.imwrite(img, 's3://bucket/img.jpg', file_client_args={
+        ...     'backend': 'petrel'})
     """
-    if auto_mkdir:
-        dir_name = osp.abspath(osp.dirname(file_path))
-        mkdir_or_exist(dir_name)
-    return cv2.imwrite(file_path, img, params)
+    assert is_filepath(file_path)
+    file_path = str(file_path)
+    if auto_mkdir is not None:
+        warnings.warn(
+            'The parameter `auto_mkdir` will be deprecated in the future and '
+            'every file clients will make directory automatically.')
+    file_client = FileClient.infer_client(file_client_args, file_path)
+    try:
+        img_ext = osp.splitext(file_path)[-1]
+        # Encode image according to image suffix.
+        # For example, if image path is '/path/your/img.jpg', the encode
+        # format is '.jpg'.
+        _, img_buff = cv2.imencode(img_ext, img, params)
+        file_client.put(img_buff.tobytes(), file_path)
+    except Exception as error:
+        warnings.warn(error)
```

Comment:
please @ZwwWayne have a look. If the `imwrite` occurred exception, is it useful to help users debug just printing warning information?

Reply:
I do not think we should handle that because we should let the exact place to replace its error rather than leave it in later with potentially misleading error messages.

Reply:
Because this interface specifies return the bool type, and it does not throw an exception, it may cause the interface inconsistent issue.

Reply:
Then should it raise a warning of error and directly return False?

Reply:
yes

Reply:
Sorry, in cv2.imwrite, if file_path extension is error, it will throw an exception. 
```python
>>> cv2.imwrite('1.jpgg', np.zeros((2,2)))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
cv2.error: OpenCV(4.5.4) /tmp/pip-req-build-3129w7z7/opencv/modules/imgcodecs/src/loadsave.cpp:728: error: (-2:Unspecified error) could not find a writer for the specified extension in function 'imwrite_'
```
I will fix it error.